Cargo.toml

```diff
@@ -102,24 +101,48 @@ lto = true
 [workspace.dependencies]
 bytes = { version = "1" }
 drain = { version = "0.1", default-features = false }
-h2 = { version = "0.3" }
-http = { version = "0.2" }
-http-body = { version = "0.4" }
-hyper = { version = "0.14.32", default-features = false }
-prost = { version = "0.12" }
-prost-build = { version = "0.12", default-features = false }
-prost-types = { version = "0.12" }
-tokio-rustls = { version = "0.26", default-features = false, features = [
-    "ring",
-    "logging",
-] }
-tonic = { version = "0.10", default-features = false }
-tonic-build = { version = "0.10", default-features = false }
-tower = { version = "0.4.13", default-features = false }
+h2 = { version = "0.4" }
+http = { version = "1" }
+http-body = { version = "1" }
+hyper = { version = "1", default-features = false }
+hyper-util = { version = "0.1", default-features = false }
+prost = { version = "0.13" }
+prost-build = { version = "0.13", default-features = false }
+prost-types = { version = "0.13" }
+tonic = { version = "0.12", default-features = false }
+tonic-build = { version = "0.12", default-features = false }
+tower = { version = "0.4", default-features = false }
 tower-service = { version = "0.3" }
 tower-test = { version = "0.4" }
 
+[workspace.dependencies.http-body-util]
+version = "0.1"
+default-features = false
+features = ["channel"]
+
+[workspace.dependencies.tokio-rustls]
+version = "0.26"
+default-features = false
+features = ["logging", "ring"]
+
+# Temporarily patch `drain`. When linkerd/drain-rs#36 is released, this can
+# be removed.
+[patch.crates-io.drain]
+git = "https://github.com/linkerd/drain-rs.git"
+rev = "7678cbbf822c922fefa5c8c62070f52bea385fc9"
```

Comment:
Doesn't this patch move us to tower 0.5? surprising!

In any case, drain 0.2.0 is available

Reply:
it does! that predates our conversation [above](https://github.com/linkerd/linkerd2-proxy/pull/3504#discussion_r1988082658), and #3744.

i've since removed this, and deferred our update to tower 0.5 for #3744. note that we _do_ still have a duplicate dependency, transitively through axum and tonic:

```
[nix-shell:~/linkerd/linkerd2-proxy]$ cargo tree -i -p tower@0.5
tower v0.5.2
└── axum v0.7.9
    └── tonic v0.12.3
    ...
``

..but otherwise, the rest of the project remains pinned to v0.4.

thanks for publishing `drain` 0.2! :pray: